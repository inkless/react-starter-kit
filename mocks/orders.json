[
  {
    "id": "9H4whD",
    "name": "Banana Split",
    "destination": "185 Culver Blvd, Playa Del Rey, CA 90293",
    "dispatchTime": 119
  },
  {
    "id": "EUtSqE",
    "name": "McFlury",
    "destination": "13455 Maxella Ave, Marina Del Rey, CA 90292",
    "dispatchTime": 98
  },
  {
    "id": "uoierx",
    "name": "Acai Bowl",
    "destination": "4720 Admiralty Way, Marina Del Rey, CA 90292",
    "dispatchTime": 35
  },
  {
    "id": "oP6QHd",
    "name": "Yogurt",
    "destination": "220 Rose Ave, Venice, CA 90291",
    "dispatchTime": 105
  },
  {
    "id": "aS4EFD",
    "name": "Chocolate Gelato",
    "destination": "2323 Wilshire Blvd, Santa Monica, CA 90403",
    "dispatchTime": 73
  },
  {
    "id": "4WVKZu",
    "name": "Cobb Salad",
    "destination": "1630 Ocean Park Blvd, Santa Monica, CA 90405",
    "dispatchTime": 13
  },
  {
    "id": "5sSNvC",
    "name": "Cottage Cheese",
    "destination": "1416 4th St, Santa Monica, CA 90401",
    "dispatchTime": 3
  },
  {
    "id": "Fv8AdV",
    "name": "Coke",
    "destination": "8115 W Manchester Ave, Playa Del Rey, CA 90293",
    "dispatchTime": 119
  },
  {
    "id": "WFG4B3",
    "name": "Snow Cone",
    "destination": "2055 Sawtelle Blvd, Los Angeles, CA 90025",
    "dispatchTime": 49
  },
  {
    "id": "woqBZv",
    "name": "Pad See Ew",
    "destination": "2906 Main St, Santa Monica, CA 90405",
    "dispatchTime": 94
  },
  {
    "id": "ZEUn6r",
    "name": "Chunky Monkey",
    "destination": "400 World Way, Los Angeles, CA 90045",
    "dispatchTime": 24
  },
  {
    "id": "tG7Mtb",
    "name": "Beef Stew",
    "destination": "2208 Sawtelle Blvd, Los Angeles, CA 90064",
    "dispatchTime": 16
  },
  {
    "id": "5kyaoG",
    "name": "Cheese",
    "destination": "1303 Westwood Blvd, Los Angeles, CA 90024",
    "dispatchTime": 63
  },
  {
    "id": "2YVKCU",
    "name": "Spinach Omelet",
    "destination": "1522 Abbott Kinney Blvd, Los Angeles, CA 90291",
    "dispatchTime": 78
  },
  {
    "id": "z57hNM",
    "name": "Beef Hash",
    "destination": "10117 Washington Blvd, Culver City, CA 90232",
    "dispatchTime": 24
  },
  {
    "id": "7FNaQi",
    "name": "Pork Chop",
    "destination": "1716 S Sepulveda Blvd, Los Angeles, CA 90025",
    "dispatchTime": 93
  },
  {
    "id": "EJyQaE",
    "name": "Kale Salad",
    "destination": "9409 Venice Blvd, Culver City, CA 90232",
    "dispatchTime": 110
  },
  {
    "id": "n5byZt",
    "name": "Fresh Fruit",
    "destination": "13455 Maxella Ave, Ste 250, Marina Del Rey, CA 90292",
    "dispatchTime": 66
  },
  {
    "id": "8yJnGe",
    "name": "Cranberry Salad",
    "destination": "101 Santa Monica Blvd, Santa Monica, CA 90401",
    "dispatchTime": 109
  },
  {
    "id": "tr5K86",
    "name": "Fudge Ice Cream Cake",
    "destination": "1321 3rd St Promenade, Santa Monica, CA 90401",
    "dispatchTime": 54
  },
  {
    "id": "UWkU5v",
    "name": "Mint Chocolate Ice Cream",
    "destination": "267 N Canon Dr, Beverly Hills, CA 90210",
    "dispatchTime": 39
  },
  {
    "id": "2ndwz7",
    "name": "Vegan Pizza",
    "destination": "3306 Lincoln Blvd, Santa Monica, CA 90405",
    "dispatchTime": 40
  },
  {
    "id": "7VD4xy",
    "name": "Orange Chicken",
    "destination": "3918 Van Buren Pl, Culver City, CA 90232",
    "dispatchTime": 112
  },
  {
    "id": "3hWiEm",
    "name": "MeatLoaf",
    "destination": "11777 San Vicente Blvd., Los Angeles, CA 90049",
    "dispatchTime": 26
  },
  {
    "id": "PQnWkG",
    "name": "Milk",
    "destination": "2310 Pico Blvd, Santa Monica, CA 90405",
    "dispatchTime": 112
  },
  {
    "id": "AinKVi",
    "name": "Pastrami Sandwich",
    "destination": "10113 Washington Blvd, Culver City, CA 90232",
    "dispatchTime": 120
  },
  {
    "id": "VrXRDR",
    "name": "Arugula",
    "destination": "2911 Main St, Santa Monica, CA 90405",
    "dispatchTime": 41
  },
  {
    "id": "sKar6C",
    "name": "Pickles",
    "destination": "210 Wilshire Blvd, Santa Monica, Santa Monica, CA 90401",
    "dispatchTime": 109
  },
  {
    "id": "7Ur6qT",
    "name": "Chicken",
    "destination": "193 Culver Blvd, Playa Del Rey, CA 90293",
    "dispatchTime": 47
  },
  {
    "id": "RoYGTg",
    "name": "Cookie Dough",
    "destination": "1323 Montana Ave, Santa Monica, CA 90403",
    "dispatchTime": 38
  }
]
