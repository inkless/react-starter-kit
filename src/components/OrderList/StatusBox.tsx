import React, { useState } from 'react';

import { grays, colors } from '@src/theme';
import Heading from '@src/components/base/Typography/Heading';
import Text from '@src/components/base/Typography/Text';
import Flex from '@src/components/base/Grid/Flex';
import List from '@src/components/base/List/List';
import ListItem from '@src/components/base/List/ListItem';
import Button from '@src/components/base/Button/Button';
import Card, { Props as CardProps } from '@src/components/base/Card/Card';

export type Order = {
  id: string;
  name: string;
  startTime?: Date;
  dispatchTime?: number;
  endTime?: Date;
};
export type Props = {
  name: string;
  orders: Order[];
  limit: number;
  moveCamera?: (id: string) => void;
} & CardProps;

/**
 * Get description based on order data
 * @param order
 */
function getDescription(order: Order) {
  let description = '';
  if (order.dispatchTime) {
    description = `dispatch in ${order.dispatchTime} seconds`;
  } else if (order.endTime && order.startTime) {
    const timeUsed = Math.round((order.endTime.valueOf() - order.startTime.valueOf()) / 60000);
    description = `delivered at ${order.endTime.toLocaleTimeString()} (~ ${timeUsed} mins)`;
  } else if (order.startTime) {
    description = `dispatch started at ${order.startTime.toLocaleTimeString()}`;
  }
  return description;
}

/**
 * Render StatusBox, showing all the order details in this status
 */
export function StatusBox({ name, orders, limit, color, ...props }: Props) {
  // when orders are too much, we want to show a showMore button
  const [showAll, setShowAll] = useState(false);
  const needShowMore = orders.length > limit;

  // when orders exceed limit and showAll is false, we only show limited orders,
  // otherwise, we show all orders
  const showingOrders = needShowMore && !showAll ? orders.slice(0, limit) : orders;

  function handleShowMoreClick() {
    setShowAll(!showAll);
  }

  function handleListItemClick(id: string) {
    return () => {
      if (props.moveCamera) {
        props.moveCamera(`current-point-${id}`);
      }
    };
  }

  return (
    <Card {...props}>
      <Heading>{name}</Heading>
      <List>
        {showingOrders.length ? (
          showingOrders.map((order, index) => (
            <ListItem key={index} onClick={handleListItemClick(order.id)}>
              <Text padding={0} fontSize={3}>
                {order.name}
              </Text>
              <Text textAlign="right" fontSize={2} padding={0} color={grays[7]}>
                {getDescription(order)}
              </Text>
            </ListItem>
          ))
        ) : (
          <Text fontSize={3} color={grays[7]}>
            No {name} orders
          </Text>
        )}
      </List>
      {needShowMore ? (
        <Flex justifyContent="center">
          <Button onClick={handleShowMoreClick} border={0} color={colors.primary.main}>
            {showAll ? 'Show Less' : 'Show More'}
          </Button>
        </Flex>
      ) : null}
    </Card>
  );
}

StatusBox.defaultProps = {
  limit: 5,
  marginBottom: 2,
  padding: 0,
  paddingTop: 2,
};

export default StatusBox;
