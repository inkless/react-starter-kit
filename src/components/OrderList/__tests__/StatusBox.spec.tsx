import React from 'react';
import { shallow } from 'enzyme';

import { orders } from './__fixture__/orders';
import StatusBox from '../StatusBox';

describe('Component StatusBox', () => {
  it('should render StatusBox with no orders', () => {
    const component = shallow(<StatusBox name="My Order" orders={[]} />);
    expect(component).toMatchSnapshot();
  });

  it('should render StatusBox with less than 5 orders', () => {
    const component = shallow(<StatusBox name="My Order" orders={orders.slice(0, 3)} />);
    expect(component).toMatchSnapshot();
  });

  it('should render StatusBox with more than 5 orders', () => {
    const component = shallow(<StatusBox name="My Order" orders={orders} />);
    expect(component).toMatchSnapshot();

    // click show more
    component.find('Button').simulate('click');
    expect(component).toMatchSnapshot();

    // click show less
    component.find('Button').simulate('click');
    expect(component).toMatchSnapshot();
  });
});
