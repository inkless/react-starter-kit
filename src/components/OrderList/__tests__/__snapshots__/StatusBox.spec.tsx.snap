// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component StatusBox should render StatusBox with less than 5 orders 1`] = `
<Styled(styled.div)
  borderRadius={2}
  boxShadow={1}
  marginBottom={2}
  padding={0}
  paddingTop={2}
>
  <Styled(Styled(styled.div))
    as="h2"
    fontSize={4}
    fontWeight={700}
    m={0}
  >
    My Order
  </Styled(Styled(styled.div))>
  <Styled(styled.div)
    as="ul"
    boxShadow={0}
    margin={0}
    paddingBottom={1}
    paddingLeft={0}
    paddingRight={0}
    paddingTop={1}
  >
    <Styled(styled.div)
      as="li"
      key="0"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Banana Split
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 119 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="1"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        McFlury
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 98 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="2"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Acai Bowl
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 35 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
  </Styled(styled.div)>
</Styled(styled.div)>
`;

exports[`Component StatusBox should render StatusBox with more than 5 orders 1`] = `
<Styled(styled.div)
  borderRadius={2}
  boxShadow={1}
  marginBottom={2}
  padding={0}
  paddingTop={2}
>
  <Styled(Styled(styled.div))
    as="h2"
    fontSize={4}
    fontWeight={700}
    m={0}
  >
    My Order
  </Styled(Styled(styled.div))>
  <Styled(styled.div)
    as="ul"
    boxShadow={0}
    margin={0}
    paddingBottom={1}
    paddingLeft={0}
    paddingRight={0}
    paddingTop={1}
  >
    <Styled(styled.div)
      as="li"
      key="0"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Banana Split
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 119 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="1"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        McFlury
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 98 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="2"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Acai Bowl
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 35 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="3"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Yogurt
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 105 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="4"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Chocolate Gelato
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 73 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
  </Styled(styled.div)>
  <Styled(styled.div)
    justifyContent="center"
  >
    <Button
      border={0}
      borderRadius={2}
      color="#6ab183"
      onClick={[Function]}
    >
      Show More
    </Button>
  </Styled(styled.div)>
</Styled(styled.div)>
`;

exports[`Component StatusBox should render StatusBox with more than 5 orders 2`] = `
<Styled(styled.div)
  borderRadius={2}
  boxShadow={1}
  marginBottom={2}
  padding={0}
  paddingTop={2}
>
  <Styled(Styled(styled.div))
    as="h2"
    fontSize={4}
    fontWeight={700}
    m={0}
  >
    My Order
  </Styled(Styled(styled.div))>
  <Styled(styled.div)
    as="ul"
    boxShadow={0}
    margin={0}
    paddingBottom={1}
    paddingLeft={0}
    paddingRight={0}
    paddingTop={1}
  >
    <Styled(styled.div)
      as="li"
      key="0"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Banana Split
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 119 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="1"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        McFlury
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 98 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="2"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Acai Bowl
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 35 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="3"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Yogurt
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 105 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="4"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Chocolate Gelato
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 73 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="5"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Cobb Salad
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 13 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="6"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Cottage Cheese
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 3 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="7"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Coke
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 119 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="8"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Snow Cone
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 49 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="9"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Pad See Ew
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 94 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
  </Styled(styled.div)>
  <Styled(styled.div)
    justifyContent="center"
  >
    <Button
      border={0}
      borderRadius={2}
      color="#6ab183"
      onClick={[Function]}
    >
      Show Less
    </Button>
  </Styled(styled.div)>
</Styled(styled.div)>
`;

exports[`Component StatusBox should render StatusBox with more than 5 orders 3`] = `
<Styled(styled.div)
  borderRadius={2}
  boxShadow={1}
  marginBottom={2}
  padding={0}
  paddingTop={2}
>
  <Styled(Styled(styled.div))
    as="h2"
    fontSize={4}
    fontWeight={700}
    m={0}
  >
    My Order
  </Styled(Styled(styled.div))>
  <Styled(styled.div)
    as="ul"
    boxShadow={0}
    margin={0}
    paddingBottom={1}
    paddingLeft={0}
    paddingRight={0}
    paddingTop={1}
  >
    <Styled(styled.div)
      as="li"
      key="0"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Banana Split
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 119 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="1"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        McFlury
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 98 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="2"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Acai Bowl
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 35 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="3"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Yogurt
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 105 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
    <Styled(styled.div)
      as="li"
      key="4"
      onClick={[Function]}
    >
      <Styled(styled.div)
        color="#484848"
        fontSize={3}
        padding={0}
      >
        Chocolate Gelato
      </Styled(styled.div)>
      <Styled(styled.div)
        color="#7a7a7a"
        fontSize={2}
        padding={0}
        textAlign="right"
      >
        dispatch in 73 seconds
      </Styled(styled.div)>
    </Styled(styled.div)>
  </Styled(styled.div)>
  <Styled(styled.div)
    justifyContent="center"
  >
    <Button
      border={0}
      borderRadius={2}
      color="#6ab183"
      onClick={[Function]}
    >
      Show More
    </Button>
  </Styled(styled.div)>
</Styled(styled.div)>
`;

exports[`Component StatusBox should render StatusBox with no orders 1`] = `
<Styled(styled.div)
  borderRadius={2}
  boxShadow={1}
  marginBottom={2}
  padding={0}
  paddingTop={2}
>
  <Styled(Styled(styled.div))
    as="h2"
    fontSize={4}
    fontWeight={700}
    m={0}
  >
    My Order
  </Styled(Styled(styled.div))>
  <Styled(styled.div)
    as="ul"
    boxShadow={0}
    margin={0}
    paddingBottom={1}
    paddingLeft={0}
    paddingRight={0}
    paddingTop={1}
  >
    <Styled(styled.div)
      color="#7a7a7a"
      fontSize={3}
    >
      No 
      My Order
       orders
    </Styled(styled.div)>
  </Styled(styled.div)>
</Styled(styled.div)>
`;
